src/mqttcm_connect.c

```diff
@@ -130,6 +133,19 @@ void mosquittoTriggerDisconnect()
 	valueChangeFlag = 0;
 }
 
+void *disconnect_thread(void *status) {
+	pthread_detach(pthread_self());
+	while (1)
+	{
+		pthread_mutex_lock(&disconnect_mutex);
+		pthread_cond_wait(&disconnect_cond, &disconnect_mutex);
```

Comment:
Add an info log here as "Received signal to disconnect , proceeding to disconnect with broker"

Reply:
added